layers/sync/sync_validation.cpp

```diff
@@ -3648,6 +3648,74 @@ void SyncValidator::PostCallRecordGetSwapchainImagesKHR(VkDevice device, VkSwapc
     }
 }
 
+// Returns null when device address is asssociated with no buffers or more than one buffer.
+// Otherwise returns a valid buffer (device address is associated with a single buffer).
+// When syncval adds memory aliasing support the need of this function can be revisited.
+static const vvl::Buffer *GetSingleBufferFromDeviceAddress(const vvl::Device &device, VkDeviceAddress device_address) {
+    vvl::span<vvl::Buffer *const> buffers = device.GetBuffersByAddress(device_address);
+    if (buffers.empty()) {
+        return nullptr;
+    }
+    if (buffers.size() > 1) {  // memory aliasing use case
+        return nullptr;
+    }
+    return buffers[0];
+}
+
+struct AccelerationStructureGeometryInfo {
+    const vvl::Buffer *vertex_data = nullptr;
+    ResourceAccessRange vertex_range;
+    const vvl::Buffer *index_data = nullptr;
+    ResourceAccessRange index_range;
+};
+
+static std::optional<AccelerationStructureGeometryInfo> GetValidGeometryInfo(
+    const vvl::Device &device, const VkAccelerationStructureGeometryKHR &geometry,
+    const VkAccelerationStructureBuildRangeInfoKHR &range_info) {
+    if (geometry.geometryType == VK_GEOMETRY_TYPE_TRIANGLES_KHR) {
+        const VkAccelerationStructureGeometryTrianglesDataKHR &triangles = geometry.geometry.triangles;
+        const uint32_t vertex_position_size = vkuGetFormatInfo(triangles.vertexFormat).texel_block_size;
```

Comment:
1. It is the logic
2. See comment in function, basically vertex formats have restrictions and we can make assumptions

```suggestion
        const uint32_t vertex_position_size = GetVertexInputFormatSize(triangles.vertexFormat);
```

Reply:
"Vertex input" this sounds it more related to vertex input stages of fixed-fucntion graphic pipeline. These stages are not used in ray tracing, so it might be a good idea not to re-use this. For example, in graphics pipeline there are rules about conversion rules of vertex attributes to the types defined by the shaders (just an example of one difference) and there is no such thing for ray tracing - it's a regular buffers that do not interract with fixed-function functionality.

I did not check the spec. If it turns out formats of the vertex data that can be used by the acceleration structure build process are related to graphics pipeline vertex input stage then it makes sense to re-use format related helpers. For now leave as it is, should be easy to update if needed.